### ISSUE TITLE
Integrated gradients for nlp models. (Explainable AI)

### ISSUE THREAD

**Issue by innat** [keras-team/keras-nlp#93]
**Describe feature**
The  [Integrated Gradients](https://www.tensorflow.org/tutorials/interpretability/integrated_gradients) (IG) can be a great tool to understand the neural network. 

**How API will change?**
It would enhance as follows:

```python

from keras_nlp.utils.visualization import IntegratedGradients as IG 

ig  = IG (model, layer=layer, ...)
```

**Candidate Solution**
- [ALIBI-Explain](https://docs.seldon.io/projects/alibi/en/latest/examples/integrated_gradients_imdb.html): It's a toolbox that provides many solutions, including **IG** in TensorFlow 2 (Keras).

Demo o/p: 
```python
ig.explain(sample)
```
![image](https://user-images.githubusercontent.com/17668390/162202861-45232f63-bb6d-4dbf-8ec7-879197c2a31f.png)

**Comment by mattdangerw** [comment]
Definitely, it would be great to start adding explainability tools here! I was actually just wishing we had precisely this while working on a guide for keras-nlp.

Would you be interested in contributing?

Also, trying to understand the design proposed more...
 - IG is it's own non-layer class?
 - What assumptions are being made about the model parameter passed in? If any.

**Comment by abheesht17** [comment]
Hey, @innat! Are you still working on this? If not, I'd like to work on this. Thanks! :)

**Comment by innat** [comment]
@abheesht17 I'm not working on it. Please, take action if you want.

**Comment by abheesht17** [comment]
Thanks, @innat! @mattdangerw, I can take this up (if we want this layer) :)
